### ISSUE TITLE
Regression: Verification failed (e=0x7e) on VKGSRender.cpp:3202

### ISSUE THREAD

**Issue by kamer1337** [RPCS3/rpcs3#6540]
reggression in Tony Hawk's Proving Ground im getting this random error, it wasnt happening 

> ·E 0:03:42.687697 {RSX [0x0117e20]} RSX: Possible data corruption on frame context storage detected
> ·F 0:03:42.687836 {RSX [0x0117e20]} RSX: class std::runtime_error thrown: Verification failed (e=0x7e): 
> (in file C:\projects\rpcs3\rpcs3\Emu\RSX\VK\VKGSRender.cpp:3202)

[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/3602834/RPCS3.zip)

**Comment by AniLeo** [comment]
Tagging @kd-11 for triaging this one, he may know what's wrong without needing to know latest build where it worked correctly

**Comment by aka2k** [comment]
I'm sorry to barge in the thread, but a very similar error is thrown on GT5 (BCES00569) with a fatal error: 

·E 0:04:35.268756 {RSX [0x0011080]} RSX: Possible data corruption on frame context storage detected
·F 0:04:35.268868 {RSX [0x0011080]} RSX: class std::runtime_error thrown: Verification failed: 
(in file C:\projects\rpcs3\rpcs3\Emu\RSX\VK\VKGSRender.cpp:3212)
·W 0:04:35.271730 {cellAudio Thread} cellAudio: Audio backend stopped unexpectedly, likely due to a buffer underrun

Sometimes it throws the very same line: 3202. Other times it's the line above: 3212. Either way, I'm attaching the log.

[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/3612097/RPCS3.zip)

**Comment by NNNAAA0123** [comment]
Sorry to barge the thread even more, but from my experience I had this error with GT5 too when multithreaded RSX was checked, I had the same error, same line (3212), and I too had error messages that mentioned audio stuff, so I just switched to openAL, but more importantly disabled multithreaded RSX.

I have a fx 8350 and what I think happen (just a theory) is that when playing the game, espacially for the first time, when the shader and LLVM recompiler are at work, my cpu was at like 100-95%, when I open my config file PPU thread is set to 2 and my max SPU thread is 6, since I only have 8 thread, if there is too much resources used by the PPU and SPU, the RSX has no resources left and that cause problems, I used to have error message mentioning desync (which most of the time meant the game just froze and the desync error got spammed endlessly) and a message mentioning that there isn't enough space in the CB chain (which happened at the same time as ~1sec froze of the game) which I think all happened because of that. That's the only explanation I got since I saw on youtube a guy having multithreaded rsx on no problems on a much more powerful cpu (r7 3700x), his rpcs3 version was older (0.0.6-8392) and this is a regression issue, I know this isn't a very scientific method of trying to explain stuff.

TL;DR: I think the errors on GT5 are caused by the mutlithreaded RSX option when the hardware is not powerful enough, (also if that's your case, Giga SPU block size, gave me more fps).
Sorry if what I said was stupid, I don't know much about rpcs3's inner workings.

**Comment by kamer1337** [comment]
i think you are right the error is related to Multithreaded RSX, turned it off and the error seems to be gone

**Comment by aka2k** [comment]
GT5 seems to be fixed by disabling Multi-threaded RSX, but GT6 needs also to be set on Safe SPU block size to be more stable again. It still soft-lock from time to time, especially when compiling shaders. Switching to OpenAL, in my case, made things much worse. So I just left that on XAudio.

**Comment by kd-11** [comment]
Anyone having this issue on a non-nvidia card? I need to narrow down the possible causes as the 'corruption' error is usually a result of the driver returning something unexpected. Its been a few years since we had this problem so something must have changed recently outside the emulator itself.

**Comment by aka2k** [comment]
> 
> 
> Anyone having this issue on a non-nvidia card? 

My rig:

i7 5775C
AMD RX 580 8Gb on the latest driver
16Gb DDR3

**Comment by kd-11** [comment]
I'm having a hard time reproducing this, I'll revisit once we have a wider selection of affected titles and more reliable steps to reproduce.

**Comment by NNNAAA0123** [comment]
fx 8350 (4ghz stock voltage), 16gb ddr3 ram (1600mhz), r9 390 (driver 19.5.2), windows 10 pro (1903), BCES00569 Game version 1.05 (though I had the same error in 1.00, and any version above 1.05 and I experience softlocks and crashes no matter the settings)
[rpcs3.zip](https://github.com/RPCS3/rpcs3/files/3623557/rpcs3.zip)

This zip file contain 2 RPCS3.log (the TTY.log files were both empty, so they're not included)
Each run I did the same thing, launched the game, and jumped into the same race with the same car, on the run where mutlithreaded RSX is off, since it didn't crash I quit the game and rpcs3 after one lap, there are probably other random elements that varies between the runs but I doubt it matter much.

Both run used the same rpcs3 settings which are:
CPU: LLVM PPU & SPU, 6 SPU Threads, liblv2.sprx only, SPU loop detection & SPU cache on everything else off, Giga SPU Block Size
GPU: Vulkan, Everything else is default
Audio: OpenAL, Everything else is default
Every other category is left to default, except for the performance overlay and the multithreaded RSX option which was activated in the file "GT5MULTIRSX.log", and deactivated in the file "GT5NOMUTLIRSX.log".

Also when I jumped in the race with the multithreaded RSX option on, I experienced stutter right away with errors saying that the CB chain has run out of entry (which I don't have with mutithreaded RSX off), and after a few seconds it crashed, from my experience it tends to happen when there is a lot of visual action on screen, which in this game are cars and particularly premium ones (which are cars that are more detailed), but I might be wrong.

**Comment by kd-11** [comment]
I still cannot get this to trigger on my end. If you're confident about easily reproducing the crash, use [this build](https://www.dropbox.com/s/7zzp5lqz5ewgnz2/rpcs3-frame-debug.7z?dl=1) and attach the log after the crash happens. It has more information added to the log file that could be helpful.

**Comment by kamer1337** [comment]
here is log from your build 
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/3629829/RPCS3.zip)

**Comment by kd-11** [comment]
I strongly suspect there is some recursion going on here. I'll update in a few hours, possibly with another debug build. At least nothing too crazy is going on.

**Comment by ESRDEV** [comment]
I think I found something that maybe help in some way.

This error only occurs when the game has update and DLCs in the "\ dev_hdd0 \ game\<GAMESERIAL>" folder I tested with Borderlands 2 GOTY [BLES01684] updated to v01.15 and not working with LLVM Recompiler but with  Interperter (Fast) game run but unstable 8~30 FPS.

Game in v01.00 running fine with LLVM Recompiler but without any DLC :(

**Comment by HHCHunter** [comment]
I also get this in GT5 as well
```

F {RSX [0x0a7cbd4]} RSX: class std::runtime_error thrown: Verification failed: 
(in file C:\projects\rpcs3\rpcs3\Emu\RSX\VK\VKTexture.cpp:85)
```

**Comment by kd-11** [comment]
@HHCHunter That is a different error. Ease up on the resolution upscaler for that one.

**Comment by kd-11** [comment]
@ESRDEV After attempting to recreate your issue - it seems we're talking about completely different things. Yes, Borderlands2 will refuse to go ingame with PPU LLVM when DLC is present, but that is not the same as this crash noted here which is caused by Multihreaded RSX option.

**Comment by kd-11** [comment]
As for the main crash, I believe I have figured out how it can happen (in theory), I'll push some experimental patches shortly. If my guess is correct, it would be triggered by either weak CPU or low memory bandwidth (DRAM) causing a race condition later on in the pipeline.

**Comment by kd-11** [comment]
Should be fixed by https://github.com/RPCS3/rpcs3/pull/6594. This is all theoretical as I have not been able to observe this problem locally, so please test and confirm.
